fern/definition/dnsrecords.yml

```diff
@@ -0,0 +1,22 @@
+types:
+  DnsRecord:
+    properties:
+      name: string
+      ttl: integer
+      type: string
+      value: string
+  DnsRecords:
+    properties:
+      a: optional<list<DnsRecord>>
+      aaaa: optional<list<DnsRecord>>
+      mx: optional<list<DnsRecord>>
+      txt: optional<list<DnsRecord>>
+      ns: optional<list<DnsRecord>>
+      cname: optional<list<DnsRecord>>
+  DnsRecordsReport:
+    properties:
+      domain: string
+      dnsRecords: DnsRecords
+      dmarcDnsRecords: DnsRecords
```

Comment:
We could also put the DMARC and DKIM records in the main `dnsRecords` field, since they are just DNS records after all. The splitoff as written is motivated by them being records for subdomains of the domain for which the report is being generated, and distinct/separate in that sense.

Reply:
Could put an enum on a more general DNSRecord Struct that holds this meaning